Document `lazyRouteManifest` option for React Router v6 and v7 (#16397)
getsentry/sentry-docs: docs/platforms/javascript/guides/react/features/react-router/v6.mdx

Now, Sentry should generate `pageload`/`navigation` transactions with parameterized transaction names (for example, `/teams/:teamid/user/:userid`), where applicable. This is only needed at the top level of your app, unlike React Router v4/v5, which required wrapping every `<Route />` you wanted parametrized.

## Static Route Manifest

> _Available in `@sentry/react` version `10.39.0` and above._

Reviewer: There's a component for this!

When using [`patchRoutesOnNavigation`](https://reactrouter.com/6.30.3/routers/create-browser-router#optspatchroutesonnavigation) to dynamically load route definitions, the full route hierarchy isn't available to Sentry until each route is navigated to. This can cause transactions to receive incomplete or wildcard names (for example, `/users/*` instead of `/users/:userId`).

To ensure accurate transaction names, you can provide a static list of route patterns via the `lazyRouteManifest` option. When provided, Sentry uses this manifest as the primary source for determining transaction names without needing to wait for route modules to load.

<Alert level="warning" title="Important">

> `lazyRouteManifest` requires `enableAsyncRouteHandlers: true` to be set on the integration.

Reviewer 2: ```suggestion
`lazyRouteManifest` requires `enableAsyncRouteHandlers: true` to be set on your `reactRouterV6BrowserTracingIntegration` configuration.
```
since this is in an alert, we should be as clear as possible

</Alert>

> 

Reviewer 2: Let's add the explanation here:

```suggestion
To use `lazyRouteManifest`, you need to set `enableAsyncRouteHandlers: true` in your `reactRouterV6BrowserTracingIntegration` configuration:
```
